compiler/rustc_target/src/spec/mod.rs

```diff
@@ -3602,6 +3602,24 @@ impl Target {
             _ => return None,
         })
     }
+
+    /// Returns whether this target is known to have unreliable alignment:
+    /// native C code for the target fails to align some data to the degree
+    /// required by the C standard. We can't *really* do anything about that
+    /// since unsafe Rust code may assume alignment any time, but we can at least
+    /// inhibit some optimizations, and we suppress the alignment checks that
+    /// would detect this unsoundness.
+    ///
+    /// Every `return true` here is still a soundness bug for that target.
+    pub fn has_unreliable_alignment(&self) -> bool {
+        // FIXME(#112480) MSVC on x86-32 is unsound and fails to properly align many types with
+        // more-than-4-byte-alignment on the stack.
+        if self.is_like_msvc && self.arch == "x86" {
```

Comment:
This function is called a lot during codegen. Should I be worried about the (short) string comparison here?

We can also add a new `bool` flag to the target spec that directly controls this.

Reply:
Hm, thinking about this some more it might make sense to make it a part of the target spec so that custom targets can be supported.

Reply:
Currently we auto-detect this for custom targets, if we make it part of the spec custom target specs will have to remember to do this.